account-kit/logging/src/client.ts

```diff
@@ -1,17 +1,60 @@
+import { AnalyticsBrowser } from "@segment/analytics-next";
+import { uuid } from "uuidv4";
+import { WRITE_IN_DEV, WRITE_KEY } from "./_writeKey.js";
+import { noopLogger } from "./noop.js";
+import { ContextAllowlistPlugin } from "./plugins/contextAllowlist.js";
+import { DevDestinationPlugin } from "./plugins/devDestination.js";
 import type { EventLogger, EventsSchema } from "./types";
 
 export function createClientLogger<
   Schema extends EventsSchema
 >(): EventLogger<Schema> {
-  return {
-    trackEvent: ({ name, data }) => {
-      if (window.location.hostname.includes("localhost")) {
-        // don't log events from localhost... hmm how do we do this while still being able to test locally?
-        // I'll come back to this once I have analytics JS setup. I think I can write events to console using analytics JS middleware
-        return;
-      }
+  const isDev = window.location.hostname.includes("localhost");
+  if (WRITE_KEY == null || (isDev && !WRITE_IN_DEV)) {
+    // If we don't have a write key, we don't want to log anything
+    // This is useful for dev so we don't log dev metrics
+    //
+    // We also don't allow logging on localhost unless WRITE_IN_DEV is set to true
+    // WRITE_IN_DEV is only ever true if you're building from source with env vars set to true
+    return noopLogger;
+  }
+
+  const analytics = AnalyticsBrowser.load(
+    {
+      writeKey: WRITE_KEY,
+      // we disable these settings in dev so we don't fetch anything from segment
+      cdnSettings: isDev
+        ? {
+            integrations: {},
+          }
+        : undefined,
+    },
+    // further we disable the segment integration dev
+    {
+      disableClientPersistence: true,
+      integrations: {
+        "Segment.io": !isDev,
+      },
+    }
+  );
```

Comment:
I think we'll probably want to setup a proxy so we can send metrics to `metrics.accountkit.alchemy.com` or something: https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/custom-proxy/

two main reasons are:
1. data destination abstraction
2. avoiding having these DII events being blocked by browsers / extensions